lib/career_bot/responders/message_responder.rb

```diff
@@ -3,32 +3,44 @@ def initialize(message)
     @message = message
   end
 
-  def set_action
-    case
-      when message.text == 'test'
-        message.reply(I18n.t('test_message'))
-      when message.attachments && conversation.apply == false
-        message.reply(I18n.t('attachment_response'))
-        message.reply(gif_response)
-      when conversation.apply
-        run_apply_process
-      when quick_reply && quick_reply != 'empty'
-        run_postback
-      else
-        send_to_wit
-    end
+  def response
+    test_message
+    set_action
   end
 
   private
 
   attr_reader :message
+
+  def set_action
+    if message_data.apply
+      run_apply_process
+    elsif message.attachments
+      gif_response
+    elsif quick_reply && quick_reply != 'empty'
+      run_postback
+    else
+      process_message
+    end
+  end
+
+  def test_message
+    return unless message.text == 'test'
```

Comment:
Why not `if`? For me it's more readable than `return unless`

Reply:
http://www.rubydoc.info/github/bbatsov/rubocop/Rubocop/Cop/Style/GuardClause

and i really like it ;)